### PR TITLE
Add ZIP 244 auth digest test vectors

### LINKED ISSUE #33: Implement ZIP 244 auth digest test vectors
This was missed from #20.

### PR DESCRIPTION
**Pull request by str4d**
Closes #33.

### CODE REVIEW

File: zip_0244.py [hunk 1]
```diff
@@ -45,6 +51,20 @@ def sapling_digest(tx):
 
     return digest.digest()
 
+def sapling_auth_digest(tx):
+    digest = blake2b(digest_size=32, person=b'ZTxAuthSapliHash')
+
+    if len(tx.vSpendsSapling) + len(tx.vOutputsSapling) > 0:
+        for desc in tx.vSpendsSapling:
+            digest.update(bytes(desc.proof))
```
Comment (nuttycom) [inline zip_0244.py:59]: Isn't `desc.proof` already a bytes value?

File: zip_0244.py [hunk 2]
```diff
@@ -45,6 +51,20 @@ def sapling_digest(tx):
...
+        for desc in tx.vSpendsSapling:
+            digest.update(bytes(desc.spendAuthSig))
+        for desc in tx.vOutputsSapling:
+            digest.update(bytes(desc.proof))
```
Comment (nuttycom) [inline zip_0244.py:63]: Same question here.

**Review by nuttycom — APPROVED** [review]
Checked this against the spec. One question about some of the `bytes()` calls.

File: zip_0244.py (hunk 1 above)
Comment (str4d) [reply zip_0244.py:59]: Not for Sapling (we have a `GrothProof` type).

File: zip_0244.py (hunk 2 above)
Comment (str4d) [reply zip_0244.py:63]: See above.

**Review by ebfull — APPROVED** [review]

**Pull request merged**